homeassistant/components/dormakaba_dkey/sensor.py

```diff
@@ -0,0 +1,67 @@
+"""Dormakaba dKey integration sensor platform."""
+from __future__ import annotations
+
+from py_dormakaba_dkey import DKEYLock
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+from .entity import DormakabaDkeyEntity
+from .models import DormakabaDkeyData
+
+BINARY_SENSOR_DESCRIPTIONS = (
+    SensorEntityDescription(
+        key="battery_level",
+        name="Battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the lock platform for Dormakaba dKey."""
+    data: DormakabaDkeyData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            DormakabaDkeySensor(data.coordinator, data.lock, description)
+            for description in BINARY_SENSOR_DESCRIPTIONS
+        ]
```

Comment:
`async_add_entities` can accept a generator.

```suggestion
        DormakabaDkeySensor(data.coordinator, data.lock, description)
        for description in BINARY_SENSOR_DESCRIPTIONS
```